### PR TITLE
feat: Add test for space point edm

### PR DESCRIPTION
**Pull request by CarloVarni**
Adds some unit test for the space point edm, with also checks on the types

Requires https://github.com/acts-project/acts/pull/3689 to compile (already merged)

### CODE REVIEW

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+  std::size_t n = 0ul;
+  for (const proxy_t& proxy : spContainer) {
+    float refX = 1.f * n;
+    float refY = 1.5f * n;
+    float refZ = 2.f * n;
+    float refCovR = 2.5f * n;
+    float refCovZ = 3.f * n;
+    float refRadius = std::hypot(refX, refY);
+    float refPhi = atan2f(refY, refX);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp:173]:
```suggestion
    float refPhi = std::atan2(refY, refX);
```

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  const SpacePoint& get_impl(std::size_t idx) const { return storage()[idx]; }
+
+  std::any component_impl(Acts::HashedString key, std::size_t /*n*/) const {
+    using namespace Acts::HashedStringLiteral;
+    switch (key) {
+      case "TopStripVector"_hash:
+      case "BottomStripVector"_hash:
+      case "StripCenterDistance"_hash:
+      case "TopStripCenterPosition"_hash:
+        return Acts::Vector3(0, 0, 0);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp:67]:
```suggestion
        return Acts::Vector3(0., 0., 0.);
```

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    BOOST_CHECK_EQUAL(proxy.varianceR(), refCovR);
+    BOOST_CHECK_EQUAL(proxy.varianceZ(), refCovZ);
+
+    const Acts::Vector3& topStripVector = proxy.topStripVector();
+    const Acts::Vector3& bottomStripVector = proxy.bottomStripVector();
+    const Acts::Vector3& stripCenterDistance = proxy.stripCenterDistance();
+    const Acts::Vector3& topStripCenterPosition =
+        proxy.topStripCenterPosition();
+
+    for (std::size_t i(0); i < 3ul; ++i) {
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp:190]:
```suggestion
    for (std::size_t i = 0; i < 3; ++i) {
```

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+  BOOST_CHECK_EQUAL(spContainerRef.size(), nExternalPoints);
+  BOOST_CHECK_EQUAL(spContainerVal.size(), nExternalPoints);
+}
+
+BOOST_AUTO_TEST_CASE(spacepoint_container_edm_functionalities) {
+  std::size_t nExternalPoints = 100;
+  SpacePointCollection externalCollection;
+  externalCollection.reserve(nExternalPoints);
+  for (std::size_t i(0); i < nExternalPoints; ++i) {
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp:117]:
```suggestion
  for (std::size_t i = 0; i < nExternalPoints; ++i) {
```
This is more common in ACTS and we try to get away from the `()` initialisation.

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+#include "Acts/Utilities/HashedString.hpp"
+#include "Acts/Utilities/Holders.hpp"
+
+#include <any>
+#include <cmath>
+#include <iterator>
+#include <ranges>
+#include <vector>
+
+#include <math.h>
```
[the remark this answers is not in the document]
Comment (CarloVarni) [reply Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp]: done

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp (hunk 1 above)
Comment (CarloVarni) [reply Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp]: done

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp (hunk 2 above)
Comment (CarloVarni) [reply Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp]: done

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp (hunk 4 above)
Comment (CarloVarni) [reply Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp]: done

File: Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp (hunk 3 above)
Comment (CarloVarni) [reply Tests/UnitTests/Core/EventData/SpacePointContainerEdmTests.cpp]: done

**Review by AJPfleger — APPROVED** [review]
lgtm

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [68cf91d23e5ce79e5060aaa8c9b110618cae73bc]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44308647) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8257877)!

**Pull request merged**